### ISSUE TITLE
RadiaCode: probably incorrect estimated_true_cps calculation.

### ISSUE THREAD

**Issue by Am6er** [sandialabs/SpecUtils#35]
1.Subject is:

https://github.com/sandialabs/SpecUtils/blob/f33fdcbe46a630831a8995b5494979c23d148a93/src/SpecFile_radiacode.cpp#L59

According references in comments: Knoll pg 122 and assumption 54 us, the code should be like this:

`const double estimated_true_cps = detected_cps/(1 - detected_cps * 54.0 / 1000000.0);`

2.Subject is a number 18600.

According developers information RadiaCode could take up to 50000 cps before overload.

**Comment by Am6er** [comment]
Just hold for a while for solving that issue, because in BecqMoni/RadiaCode xml format will be new attribute called "LifeTime".
I'll come back with clarifications.

**Comment by Am6er** [comment]
Update.
In new version of (https://github.com/Am6er/BecqMoni/commit/dec4d046b873c18bfc3c094328b943b827e6a6dd) there is a new parameter: 
`<LiveTime>27732.63217</LiveTime>`

Example:
[RC_Example.zip](https://github.com/user-attachments/files/16910039/RC_Example.zip)

RadiaCode developers also confirmed that they will be use this parameter.

So, there is two versions of xml format:
- with LiveTime
- w/o LiveTime

Assuming deadtime for RadiaCode equals 5 us (information from developers), calculation should be like this:
```
total_counts = 194318
real_time = 1821
dead_time = 5.0E-06
detected_cps = total_counts / real_time;
estimated_true_cps = (detected_cps)/(1 - detected_cps * dead_time)
live_time = real_time * (detected_cps / estimated_true_cps)
```

**Comment by wcjohns** [comment]
Thank you for this @Am6er !

The quoted:
  `const double estimated_true_cps = detected_cps/(1 - detected_cps * 54.0 / 1000000.0);`
looks to be equivalent to what was in the code:
  `const double estimated_true_cps = (18691.6)/(18691.6/detected_cps - 1);`
just a slightly different form.

I wonder why my measurements showed 54 us?
But anyway, I'll update the code to 5 us - thanks for finding this out - I'm sure its a much more reliable number than before.

I'll also add looking for the `<LiveTime>` element in the XML, and preferably use that to the estimated value (because maybe the detector isn't totally nonparalyzable, or there are pile-up detectors, processing limits, or other things going on).

again, thank you!
  -will

**Comment by ckuethe** [comment]
These might still be a little bit off, but here's what I measured with my RC-103 (will need to repeat with RC-102) and a few credit-card sized sources. I 3d printed a jig to hold the sources in a fixed and repeatable orientation, with the center spot of the ☢️ right over the ➕ on the bottom. 7.5-8mm from source to center of scintillator crystal.

All source measurements were at least 120s in length, background was 30min.


```
Computing deadtime from a=dtx_ba133.xml, b=dtx_cs137.xml, ab=dtx_ba133_cs137.xml
{'a': 9481.8, 'b': 3145.6, 'ab': 11183.783333333333, 'bg': 3.755}
dead time at 11183.8cps: 28.8us
lost count/sec: 1443.6 loss_fraction: 11.4%

Computing deadtime from a=dtx_ba133.xml, b=dtx_co60.xml, ab=dtx_ba133_co60.xml
{'a': 9481.8, 'b': 1830.1166666666666, 'ab': 10593.6, 'bg': 3.755}
dead time at 10593.6cps: 23.6us
lost count/sec: 718.3 loss_fraction: 6.4%

Computing deadtime from a=dtx_ba133.xml, b=dtx_eu152.xml, ab=dtx_ba133_eu152.xml
{'a': 9481.8, 'b': 5688.922535211268, 'ab': 13283.04132231405, 'bg': 3.755}
dead time at 13283.0cps: 20.2us
lost count/sec: 1887.7 loss_fraction: 12.4%

Computing deadtime from a=dtx_co60.xml, b=dtx_cs137.xml, ab=dtx_co60_cs137.xml
{'a': 1830.1166666666666, 'b': 3145.6, 'ab': 4341.358333333334, 'bg': 3.755}
dead time at 4341.4cps: 63.8us
lost count/sec: 634.4 loss_fraction: 12.7%

Computing deadtime from a=dtx_co60.xml, b=dtx_eu152.xml, ab=dtx_co60_eu152.xml
{'a': 1830.1166666666666, 'b': 5688.922535211268, 'ab': 6708.6, 'bg': 3.755}
dead time at 6708.6cps: 45.9us
lost count/sec: 810.4 loss_fraction: 10.8%

Computing deadtime from a=dtx_eu152.xml, b=dtx_cs137.xml, ab=dtx_cs137_eu152.xml
{'a': 5688.922535211268, 'b': 3145.6, 'ab': 7988.166666666667, 'bg': 3.755}
dead time at 7988.2cps: 26.4us
lost count/sec: 846.4 loss_fraction: 9.6%

```

**Comment by Am6er** [comment]
Interesting measurements. As far as I understand the method of determining the dead time itself from two sources, cps A+B should lead to a significant degradation of the detector measurement. Assuming, that RadiaCode built-in overload threshold is 50000 cps, we need sources that will give up at least 40000 cps. Unfortunately, I don't have such sources :(

Btw why is there such a large variation in the dead time calculation results? What is the reason for this?

**Comment by ckuethe** [comment]
Not sure.

Another data point, this time from my RC-102. I have an old, radium-powered, ship light. Call it source 'a', unsure of its age or original activity but in the fixture I put together it was reading about 8kcps. I then took all my credit card sources (Ba-133, Co-57, Co-60, Cs-137, Eu-152, Na-22) and stacked them together as source 'b'. Nominally they're 1uCi and about 1 year old, so the Na-22 and Co-57 will have decayed a bunch.

This is about the hottest combination I can cook up, unless I start buying and disassembling old Pyrotronics smoke detectors, or unless someone drops a few 10uCi sources in my mailbox.

```
Computing deadtime from a=a.xml, b=b.xml, ab=ab.xml
{'a': 8028.616666666667, 'b': 10250.033333333333, 'ab': 15035.233333333334, 'bg': 4.777777777777778}
dead time at 15035.2cps: 24.0us
lost count/sec: 3243.4 loss_fraction: 17.7%
```

**Comment by wcjohns** [comment]
Thanks for adding those measurements Chris!

I recall there being some source-dependance when I tried to find dead time as well - but didn't have the time to fully investigate.  My guess was that it is an energy dependence thing, but it could also maybe be different hardware effects manifesting at different rates, that I don't understand.  From your measurements Chris, it doesn't seem to be so-much a rate thing either.

It's interesting all the dead times are ~5 to ~10 times what we should expect.

A minor side note: for the measurements with the sources almost touching the detector, we should also take into account cascade decays, as this will approach ~50% for Co60 on contact - but this doesn't seem like it would explain the discrepancy.  I think, but can't find my notes, that my measurements where for sources at 15 cm (Cs137) and 40 cm (Co57), so I don't think cascade-decays would effect my numbers, so didn't calculate corrections.

**Comment by ckuethe** [comment]
I figured that my stack of sources would be ... questionable at best. I didn't have a way to measure the attenuation of the 6 cards I used (no blank cards made of the same plastic but without the radioisotope), nor did I do much of anything else to conduct a reasonable experiment beyond trying to keep the physical arrangement of sources consistent. This was simply trying to see what was the highest count rate I could observe.

Last time I looked, $100/uCi seemed to be about the upper bound of the retail price for uncalibrated, exempt gamma source; I might be a bit silly at times but I'm definitely not spending $2000 on a pair of 10uCi Ba-133 sources. Nor do I know anyone who can hook me up with a gram of radium (yes, I read "The Radium Girls").
